contracts/vault/Vault.sol

```diff
@@ -118,13 +56,7 @@ contract Vault is IVault {
         Reward calldata reward,
         address funder,
         IPermit permit
-    )
-        external
-        payable
-        onlyPortal
-        onlyFundable(status)
-        returns (bool fullyFunded)
-    {
+    ) external payable onlyPortal returns (bool fullyFunded) {
         if (status == Status.Funded) {
```

Comment:
Move this to `IntentSource` as well and get rid of `status` as a param.

Reply:
Also, because things like this, I kinda prefer having a modifier instead of internal function for the validation.

You can do
```
if (...) {
  revert;
}

if (...) {
  revert;
}

if (...) {
  return;
}

_:
```
To not only validate, but skip execution in some cases.